cloudinit/sources/DataSourceEc2.py

```diff
@@ -731,28 +735,92 @@ def convert_ec2_metadata_network_config(network_md, macs_to_nics=None,
        not provided, get_interfaces_by_mac is called to get it from the OS.
     @param: fallback_nic: Optionally provide the primary nic interface name.
        This nic will be guaranteed to minimally have a dhcp4 configuration.
+    @param: apply_network_config: Boolean set True to configure all networking
+       presented by IMDS. This includes rendering secondary IPv4 and IPv6
+       addresses on all NICs and rendering network config on secondary NICs.
+       If False, only the primary nic will be configured and only with dhcp
+       (IPv4/IPv6).
 
-    @return A dict of network config version 1 based on the metadata and macs.
+    @return A dict of network config version 2 based on the metadata and macs.
     """
-    netcfg = {'version': 1, 'config': []}
+    netcfg = {'version': 2, 'ethernets': {}}
     if not macs_to_nics:
         macs_to_nics = net.get_interfaces_by_mac()
+    if not apply_network_config:
+        LOG.debug(
+            'Skipping network configuration for secondary NICs and IPs.'
+            'Datasource apply_network_config value is set False')
     macs_metadata = network_md['interfaces']['macs']
-    for mac, nic_name in macs_to_nics.items():
+    nic_idx = 1
```

Comment:
I'd like to separate our two main scenarios (apply_network_config True or False).

```
if not apply_network_config:                                                   
    # return net config for primary nic only                                   
    # dhcp4 + dhcp6 if ipv6s is non-empty                                      
    return netcfg_for_fallback_nic_only                                        
                                                                               
# here, just configure all according to the metadata                           
for mac, nic_name in sorted(macs_to_nics.items()):                             
    # no need for apply_network_config logic   
```

Reply:
separation added in 1f35018da2abd253561da82bdc7f577eb1058cb2.